spec/lib/atmosphere_spec.rb

```diff
@@ -58,4 +58,38 @@ def register_cloud_cient
       expect(Atmosphere.metrics_store).to eq 'other_client'
     end
   end
-end
\ No newline at end of file
+
+  context 'nic provider' do
+
+    class DummyNicProvider
+      def initialize(_config = nil); end
+
+      def get(_appl, _tmpl)
+        nil
+      end
+    end
+
+    context 'when compute site does not define provider class name' do
+      let(:cs) { create(:compute_site) }
+      it 'returns NullNicProvider' do
+        expect(Atmosphere.nic_provider(cs).class).
+          to eq Atmosphere::NicProvider::NullNicProvider
+      end
+    end
+
+    context 'when compute site defines provider class name' do
+      let(:c_name) { 'DummyNicProvider' }
+      let(:cs) { create(:compute_site, nic_provider_class_name: c_name) }
+      it 'returns provider of appropriate class' do
+        expect(Atmosphere.nic_provider(cs).class).to eq DummyNicProvider
+      end
+
+      it 'creates provider with compute site specific configuration' do
+        PROVIDER_CONF = 'Nic provider configuration'
+        cs.nic_provider_config = PROVIDER_CONF
+        expect(DummyNicProvider).to receive(:new).with PROVIDER_CONF
+        Atmosphere.nic_provider(cs)
+      end
+    end
```

Comment:
I've added a validation to ComputeSite model so if the DB is not corrupted class name should be valid. Do you think it is good enough and solves the problem of invalid class name? If I'm not asked explicitly to handle invalid class name in Atmosphere code I won't do more :)